### ISSUE TITLE
AbstractMethodError with Kotlin 1.3.70 (released today)

### ISSUE THREAD

**Issue by autonomousapps** [bnorm/kotlin-power-assert#8]
Please see the exception from this build scan:

https://scans.gradle.com/s/zabxwgoutrfrc/failure?openFailures=WzBd&openStackTraces=WzFd#top=0

Relevant configuration code:
```
plugins {
    id("java-gradle-plugin")
    id("com.gradle.plugin-publish") version "0.10.1"
    id("org.jetbrains.kotlin.jvm") version "1.3.70"
    `kotlin-dsl`
    id("com.bnorm.power.kotlin-power-assert") version "0.2.0"
}

tasks.withType<KotlinCompile>().matching {
    // compileTestKotlin, compileFunctionalTestKotlin, ...
    // useIR leads to compilation failures for non-standard test source sets :'(
    //it.name.endsWith("TestKotlin")
    it.name == "compileTestKotlin"
}.configureEach {
    kotlinOptions {
        // For use with the "com.bnorm.power.kotlin-power-assert" plugin, enabling power asserts in tests
        // https://github.com/bnorm/kotlin-power-assert
        useIR = true
    }
}
```

**Comment by TWiStErRob** [comment]
The actual error is here: https://scans.gradle.com/s/zabxwgoutrfrc/console-log?task=:compileTestKotlin

**Comment by TWiStErRob** [comment]
Hmm, that's fun:
https://github.com/JetBrains/kotlin/blob/v1.3.61/compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/extensions/IrGenerationExtension.kt#L17-L21
```kotlin
    fun generate(
        file: IrFile,
        backendContext: BackendContext,
        bindingContext: BindingContext
    )
```

https://github.com/JetBrains/kotlin/blob/v1.3.70/compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/extensions/IrGenerationExtension.kt#L33-L36
```kotlin
    fun generate(
        moduleFragment: IrModuleFragment,
        pluginContext: IrPluginContext
    )
```

How does one override two interface methods in binary jar?

**Comment by JakeWharton** [comment]
Just write both methods and only mark one as 'override'

On Tue, Mar 3, 2020, at 8:33 PM, Róbert Papp wrote:

> Hmm, that's fun:
> https://github.com/JetBrains/kotlin/blob/v1.3.61/compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/extensions/IrGenerationExtension.kt#L17-L21

>     fun generate(
        file: IrFile,
        backendContext: BackendContext,
        bindingContext: BindingContext
    )
> https://github.com/JetBrains/kotlin/blob/v1.3.70/compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/extensions/IrGenerationExtension.kt#L33-L36

>     fun generate(
        moduleFragment: IrModuleFragment,
        pluginContext: IrPluginContext
    )

**Comment by bnorm** [comment]
After figuring out the changes required to support 1.3.70, I don't think there is going to be an easy way to support both 1.3.60 and 1.3.70. Beyond the difference in IrGenerationExtension, the `descriptor` property move from `IrFunction` to `IrFunctionSymbol` for example.

At this point it's just not worth investing in backwards compatibility until the IR plugin API stabilizes in Kotlin.

**Comment by bnorm** [comment]
I just released version `0.3.0` with support for (and only for) Kotlin `1.3.70`.
